py/server/deephaven/experimental/iceberg.py

```diff
@@ -1009,10 +1022,30 @@ def adapter(
     try:
         return IcebergCatalogAdapter(
             _JIcebergTools.createAdapter(
-                name,
-                j_hashmap(properties if properties else {}),
-                j_hashmap(hadoop_config if hadoop_config else {}),
+                catalog_options
             )
         )
     except Exception as e:
         raise DHError(e, "Failed to build Iceberg Catalog Adapter") from e
+
+
+def _build_catalog_options(
+        name: Optional[str] = None,
+        properties: Optional[Dict[str, str]] = None,
+        hadoop_config: Optional[Dict[str, str]] = None,
+        enable_property_injection: bool = True
+):
+    try:
+        builder = _JBuildCatalogOptions.builder()
+
+        if name:
+            builder.name(name)
+
+        builder.putAllProperties(j_hashmap(properties if properties else {}))
+        builder.putAllHadoopConfig(j_hashmap(hadoop_config  if hadoop_config else {}))
+        builder.enablePropertyInjection(enable_property_injection)
+
+        return builder.build()
+
+    except Exception as e:
+        raise DHError(e, "Failed to build catalog options") from e
```

Comment:
```suggestion
        raise DHError(e, "Failed to build catalog options") from e
        
```
No newline at the end of file